### PR TITLE
unified code style for the idea project

### PR DESCRIPTION
**Pull request by celaus**

### CODE REVIEW

File: es/upstream [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit fe20ffbee4738401e966d13ac989bfa66f8e48b2
+Subproject commit 5c9916caeed49aa4cf2c9bd98638cc381697527c
```
Comment (dobe) [inline es/upstream]: i think this submodule update is not by intent, right?

File: build.gradle [hunk 2]
```diff
@@ -193,6 +193,9 @@ idea {
         ipr {
             withXml { provider ->
                 def node = provider.asNode()
+                node.append(new XmlParser().parse(file("codeStyle.xml")))
+
```
Comment (dobe) [inline build.gradle]: whitespace

File: build.gradle [hunk 3]
```diff
@@ -204,6 +207,7 @@ idea {
     }
 }
 
+
```
Comment (dobe) [inline build.gradle]: whitespace

File: codeStyle.xml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+<component name="ProjectCodeStyleSettingsManager">
+  <option name="PER_PROJECT_SETTINGS">
+    <value>
+      <option name="RIGHT_MARGIN" value="120"/>
+      <option name="INDENT_SIZE" value="4"/>
+      <option name="CONTINUATION_INDENT_SIZE" value="4"/>
+      <option name="TAB_SIZE" value="4"/>
+      <option name="USE_TAB_CHARACTER" value="false"/>
+      <codeStyleSettings language="JAVA">
+        <option name="KEEP_LINE_BREAKS" value="true"/>
```
Comment (dobe) [inline codeStyle.xml]: please check what changes, when you do a formatting of existing code. it should not be too invasive. do you plan to follow up with a reformatting PR?

**Comment by mfussenegger** [comment]
Would it make sense to use http://editorconfig.org/  instead for those kinds of settings?

**Pull request merged**